standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2367,10 +2367,21 @@ public Table getTable(String catName, String dbName, String tableName,
     return getTable(catName, dbName, tableName, validWriteIdList, false, null);
   }
 
+  @Override
+  public Table getTable(String catName, String dbName, String tableName,
```

Comment:
It makes sense to just use  GetTableResult get_table_req(1:GetTableRequest req) throws (1:MetaException o1, 2:NoSuchObjectException o2) and redirect all the other variations to use this method, so that you don't have to add tableId in all the other variations. Ideally we should deprecate and remove all the methods that doesn't take Request object.

Reply:
+1 on this.

Reply:
@kishendas  @adesh-rao Changed all methods signatures to request model